File: app/src/main/java/me/tagavari/airmessage/redux/ReduxEventTextImport.java
package me.tagavari.airmessage.redux;

import java.util.Collection;

import me.tagavari.airmessage.messaging.ConversationInfo;
import me.tagavari.airmessage.messaging.ConversationItem;

//An event to represent the status of a mass retrieval
public abstract class ReduxEventTextImport {
	public static class Start extends ReduxEventTextImport {
		private final int itemCount;
		
		public Start(int itemCount) {
			this.itemCount = itemCount;
		}
		
		public int getItemCount() {
			return itemCount;
		}
	}
	
	public static class Progress extends ReduxEventTextImport {
		private final int receivedItems, totalItems;
		
		public Progress(int receivedItems, int totalItems) {
			this.receivedItems = receivedItems;
			this.totalItems = totalItems;
		}
		
		public int getReceivedItems() {
			return receivedItems;
		}
		
		public int getTotalItems() {
			return totalItems;
		}
	}
	
	public static class Complete extends ReduxEventTextImport {
		private final Collection<ConversationInfo> conversations;
		
		public Complete(Collection<ConversationInfo> conversations) {
			this.conversations = conversations;
		}
		
		public Collection<ConversationInfo> getConversations() {
			return conversations;
		}
	}
	
	public static class Fail extends ReduxEventTextImport {
	}
}

File: common/me/tagavari/airmessage/redux/ReduxEventTextImport.kt
package me.tagavari.airmessage.redux

import me.tagavari.airmessage.messaging.ConversationInfo

//An event to represent the status of a mass retrieval
abstract class ReduxEventTextImport {
	class Start(val itemCount: Int) : ReduxEventTextImport()
	class Progress(val receivedItems: Int, val totalItems: Int) : ReduxEventTextImport()
	class Complete(val conversations: Collection<ConversationInfo>) : ReduxEventTextImport()
	class Fail : ReduxEventTextImport()
}